### PR TITLE
feat: use main for default git branch

### PR DESCRIPTION
**Pull request by ellisio**
## Hey, I just made a Pull Request!

It has become common practice to use `main` as the default branch when initializing a Git repository. This updates the GitHub integration to expose the `defaultBranch` config property. If not provided, `master` will still be used. For example:

```
integrations:
  github:
    - host: github.com
      defaultBranch: main
      apps:
        - $include: github-credentials.yaml
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jhaals** [comment]
I'm onboard making this the default, however I'm a bit hesitant to changing it without supplying a way to configure the name of the main branch in case there's tools that rely on it still being master or for that reason if people want to name it develop or something else.

**Comment by ellisio** [comment]
> I'm onboard making this the default, however I'm a bit hesitant to changing it without supplying a way to configure the name of the main branch in case there's tools that rely on it still being master or for that reason if people want to name it develop or something else.

@jhaals I’ll modify this to expose a configuration setting after the long weekend. 🍻

**Comment by ellisio** [comment]
@jhaals I have updated the PR to expose a `defaultBranch` config property for GitHub, which defaults to `master`.

File: packages/integration/src/github/config.ts [hunk 1]
```diff
@@ -65,6 +65,13 @@ export type GitHubIntegrationConfig = {
    * If no apps are specified, token or anonymous is used.
    */
   apps?: GithubAppConfig[];
+
+  /**
+   * The default branch for scaffoled repositories.
```
Comment (adamdmharvey) [inline packages/integration/src/github/config.ts:70]:
```suggestion
   * The default branch for scaffolded repositories.
```

**Comment by ellisio** [comment]
@adamdmharvey Fixed the typo, thanks! 🍻

**Comment by jhaals** [comment]
Thanks for the update @ellisio!

I realise that we currently leave our other integrations in the cold by implementing this just for GitHub so we probably want to make this available for the other VCS providers as well. However let's get GitHub in a mergeable state before proceeding with the others as it should be the same pattern.

The config option should probably reside under `scaffolder:` instead of integration. The tricky bit is that that part of the config is not passed to the `createGithubPublishAction` so you have to make it optional for this to be a non breaking change.

The alternative is to make this configurable in the template which I guess is a feature in itself.

This can be done by

Adding an optional defaultBranch to createTemplateAction https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts#L47

and defaulting it to master further down https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts#L123

**Comment by ellisio** [comment]
@jhaals Awesome, I'll cook something up when I get some time in the next day or so. 🍻

**Comment by freben** [comment]
Looking forward to it! :)

**Comment by ellisio** [comment]
Sorry for the delay folks, got retasked to standing up a new infrastructure that will be the target of our services in Backstage. So have not had a lot of time to work on this. If someone wants to jump in, feel free!

**Comment by shoukoo** [comment]
Hey there, I also need this feature and I would like to jump in and finish this off.

**Comment by shoukoo** [comment]
This is the PR - https://github.com/backstage/backstage/pull/6195

**Pull request closed without merging**